### PR TITLE
Add machine name to queue name examples

### PR DESCRIPTION
**Pull request by WilliamBZA**
Connects to https://github.com/Particular/docs.particular.net/issues/2734

This should make it clearer for users that when configuring these features for use with ServiceControl that the queues should be remote queues instead of using the Physical Routing file.

I'm a little unsure of how this looks if the users are using different transports however. We may need a better way to represent this instead of simply assuming MSMQ in documentation.

Ping @Particular/servicecontrol-maintainers please review. I suppose @Particular/nservicebus-maintainers should also review since it touches some of their pages?

### CODE REVIEW

**Comment by bording** [comment]
@WilliamBZA I'm 👎 to this change because this is an MSMQ-specific thing that isn't generally true for other transports.

**Comment by WilliamBZA** [comment]
> MSMQ-specific thing that isn't generally true for other transports.

I agree.

Any suggestions on how to make it generic while still conveying the distinction we've created for MSMQ between remote queues and the physical routing file?

**Comment by SimonCropp** [comment]
given historically we actually went through and removed all the msmq assumptions from queue names i would prefer not to put them back :)

**Comment by WilliamBZA** [comment]
Once again: I agree.

Any suggestions on how/where to put it then? We have an arbitrary "use physical routing file here, but not here" issue and we need to make it clear to users when to use each.

**Comment by weralabaj** [comment]
Instead of replacing the existing snippets could we add extra lines to them for MSMQ? Or add separate snippets?

An alternative could be "Configuring MSMQ production environment" or sth like that, and link to it.


@SimonCropp @bording

**Comment by WilliamBZA** [comment]
I was thinking about it some more today, and I think your suggestion is a good one @weralabaj.

This config is mostly specific to configuring ServiceControl, so a dedicated heading in the configuration section of the ServiceControl docs might be a good fit. I'll change the PR.

**Comment by WilliamBZA** [comment]
Still busy on this, just haven't found the time yet.

**Comment by WilliamBZA** [comment]
@Particular/docs-maintainers I'm not sure if I'm doing the snippets and partials correctly here, can somebody please verify that I'm on the correct track?

I'll add heartbeats and auditing next if the error queue config looks correct.

**Comment by weralabaj** [comment]
@WilliamBZA seems like the underlying dependency here is Core, so if I understand correctly you should place those snippets here https://github.com/Particular/docs.particular.net/tree/master/Snippets/Core. Also the component in the doco page should be core. There's nothing depending on SC version here, correct?

If you'd like to sue SC versions then that would be similar to what was done for Heartbeats here https://github.com/Particular/docs.particular.net/tree/master/Snippets/Heartbeats.

**Comment by SzymonPobiega** [comment]
Partials look ok. @weralabaj would snippets in this form work correctly?

**Comment by WilliamBZA** [comment]
@weralabaj so I should move the MSMQ specific snippets into the core sln?

**Comment by weralabaj** [comment]
sorry, I forgot the context of that issue, \docs.particular.net\Snippets\MsmqTransport probably would be even better.

**Comment by SimonCropp** [comment]
i dont think u can use msmq snippets in an SC page

**Comment by weralabaj** [comment]
at the moment there's no component specified, but once we move msmq out of the core it won't work otherwise, will it?

**Comment by WilliamBZA** [comment]
So what's the consensus on where the snippets should live? I'm confused.

**Comment by weralabaj** [comment]
@WilliamBZA I propose this - ensure you have all the content you want, and I'll sync with Simon regarding where it should live and move it accordingly, OK?

**Comment by weralabaj** [comment]
Just let me know if that's ready :D

File: Snippets/ServiceControl/ServiceControl_All/Core_[,6.0]/ErrorQueueMsmqConfigurationProvider.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿#pragma warning disable 618
```
Comment (weralabaj) [inline Snippets/ServiceControl/ServiceControl_All/Core_[,6.0]/ErrorQueueMsmqConfigurationProvider.cs]: should that be v6 inclusive or exclusive? hard to figure out which versions it applies to
also, is anything below v5 supported by servicecontrol?

@WilliamBZA ☝️

File: Snippets/ServiceControl/ServiceControl_All/Core[6.0,)/ConfigErrorQueue.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    class ConfigErrorQueue
+    {
+        Usage(EndpointConfiguration endpointConfiguration)
+        {
+            #region ConfigMsmqErrorWithCode
+
+            endpointConfiguration.SendFailedMessagesTo("targetErrorQueue@machinename");
+
+            #ConfigMsmqErrorWithCode
```
Comment (SimonCropp) [inline Snippets/ServiceControl/ServiceControl_All/Core[6.0,)/ConfigErrorQueue.cs]: this wont compile needs to be an end region

File: Snippets/ServiceControl/ServiceControl_All/Core[6.0,)/ConfigErrorQueue.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿namespace ServiceControl.ConfigureEndpoints
+{
+    using NServiceBus;
+
+    class ConfigErrorQueue
+    {
+        Usage(EndpointConfiguration endpointConfiguration)
```
Comment (SimonCropp) [inline Snippets/ServiceControl/ServiceControl_All/Core[6.0,)/ConfigErrorQueue.cs]: this wont compile as it is missing a return type or needs to be a constructor

File: servicecontrol/configuring-queues.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ - nservicebus/recoverability/configure-error-handling
+---
+
+ServiceControl reads messages from the input queue specified in the [ServiceControl instance transport configuration](creating-config-file#transport). The queues specified during this stage are populated with messages from each of the production endpoints after which, ServiceControl ingests those messages in order to extract the required information from them. Depending on the transport selected and the deployment scenario, the production endpoints might need to be configured to send the messages to a remote queue.
+
+Of the currently supported transports, only MSMQ requires special configuration to deal with remote queues.
+
+### MSMQ
+
+If the endpoints are on a separate machine to the ServiceControl instance, the endpoints must be configured to send their error and audit messages (if auditing is enabled) to a remote queue as the MSMQ instance mapping file is not used in the case of error and audit queues
```
Comment (SimonCropp) [inline servicecontrol/configuring-queues.md]: all this content and snippets needs to be moved to the MSMQ doco and linked to from here

Comment (SimonCropp) [inline servicecontrol/configuring-queues.md]: if u are talking about audit as well you need to have snippets for audit

**Comment by SimonCropp** [comment]
does anyone want to resurrect this?

**Comment by weralabaj** [comment]
I dropped the ball, will look into this today.

**Comment by WilliamBZA** [comment]
This is still blocked by me. I'm happy to close this PR completely until I can grab it again.

**Comment by weralabaj** [comment]
@WilliamBZA I can help with splitting, I think nothing related to msmq content should surprise me now after splitting some docs ;) Generally the idea was to move the MSMQ-specific content to another page, that will be component: MSMQTransport and there have all the snippets.

That will address the issue of versioning. Mike suggested we should cover only supported versions, i.e. v5 and above.

I think that should be enough for me to progress or is there anything you'd like to add/modify here first?

**Comment by WilliamBZA** [comment]
That'd be great @weralabaj, thanks for jumping on the grenade here!

**Comment by weralabaj** [comment]
@WilliamBZA oI'm not sure if my understanding is correct, so if MSMQ endpoints are on a different machine than SC then you need to configure your endpoints to send errors and audits to the queue on the same machine that SC is running on?

Can you see if snippets make sense?

**Comment by weralabaj** [comment]
@SimonCropp any idea on what I'm doing wrong? that snippet is there for msmq v1

**Comment by WilliamBZA** [comment]
@weralabaj Yes, that's exactly it. The instance mapping file isn't used for error forwarding, so we have to have different ways of configuring queus depending on whether it's a regular "business" message, or if it's a "servicecontrol" message.

Snippets look correct so far.

**Comment by WilliamBZA** [comment]
Thanks for the help @SimonCropp and @weralabaj. I think this can be merged now. I'll raise a separate issue to add the snippets for audits so we don't lose that.

**Pull request merged**